src/cpp/session/projects/SessionProjects.cpp

```diff
@@ -127,44 +127,49 @@ Error getProjectFilePath(const json::JsonRpcRequest& request,
    return Success();
 }
 
-Error findProjectInFolder(const json::JsonRpcRequest& request,
-                          json::JsonRpcResponse* pResponse)
+Error findExistingProjFileInFolder(const std::string& inFolder, std::string* pResult)
```

Comment:
I'll rename, agree it is confusing. For returning non-Success when not found, that's a bit fuzzier to me, it seems like we tend to use Error return type like exceptions; things that indicate an "exceptional" problem, versus something that can happen in normal situations (i.e. didn't find a .Rproj file). So was using an empty result to indicate that. Neither way seems 100% ideal.

Reply:
Since the method currently always returns Success(), I could switch to having it return a bool to indicate if something was found, along with the empty output string, or return the string directly (in either case an empty result still mean not-found).